app/api/route/schemas/user_schemas.py

```diff
@@ -0,0 +1,53 @@
+from typing import Any
+
+import marshmallow
+from apiflask import fields
+from marshmallow import fields as marshmallow_fields
+
+from api.db.models.user_models import Role, RoleType
+
+##############
+# Role Models
+##############
+
+
+class RoleSchema(marshmallow.Schema):
+    type = marshmallow_fields.Enum(RoleType, description="The name of the role", by_value=True)
+
+    # Output only fields
+    created_at = fields.DateTime(dump_only=True)
+    updated_at = fields.DateTime(dump_only=True)
+
+    @marshmallow.post_load
+    def make_role(self, data: dict, **kwargs: dict[str, Any]) -> Role:
+        return Role(**data)
+
+
+##############
+# User Models
+##############
+
+
+class UserSchema(marshmallow.Schema):
+    user_id = fields.UUID(dump_only=True)
+    first_name = fields.String(description="The user's first name", required=True)
+    middle_name = fields.String(description="The user's middle name")
+    last_name = fields.String(description="The user's last name", required=True)
+    phone_number = fields.String(
+        description="The user's phone number",
+        example="123-456-7890",
+        required=True,
+        pattern=r"^([0-9]|\*){3}\-([0-9]|\*){3}\-[0-9]{4}$",
+    )
+    date_of_birth = fields.Date(
+        description="The users date of birth",
+        required=True,
+    )
+    is_active = fields.Boolean(
+        description="Whether the user is active",
+        required=True,
+    )
```

Comment:
Yeah I also am wondering whether we should come up with other examples that might be more realistic, since I think it's also not common on our projects to have the client set the role. But let's do that for another PR